samples/extensions/dynamic_rendering_local_read/dynamic_rendering_local_read.cpp

```diff
@@ -0,0 +1,1052 @@
+			color_attachment_info[j].clearValue  = clear_values[j];
+		}
+
+		color_attachment_info[0].imageView = swapchain_buffers[i].view;
+		for (auto i = 0; i < 3; i++)
+		{
+			color_attachment_info[i + 1].imageView = attachment_list[i].view;
+		}
+
+		VkRenderingAttachmentInfoKHR depth_attachment_info = vkb::initializers::rendering_attachment_info();
+		depth_attachment_info.imageView                    = depth_stencil.view;
+		depth_attachment_info.imageLayout                  = VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL;
+		depth_attachment_info.resolveMode                  = VK_RESOLVE_MODE_NONE;
+		depth_attachment_info.loadOp                       = VK_ATTACHMENT_LOAD_OP_CLEAR;
+		depth_attachment_info.storeOp                      = VK_ATTACHMENT_STORE_OP_DONT_CARE;
+		depth_attachment_info.clearValue                   = clear_values[1];
+
+		VkRenderingInfoKHR render_info   = vkb::initializers::rendering_info();
+		render_info.renderArea           = {0, 0, width, height};
+		render_info.layerCount           = 1;
+		render_info.colorAttachmentCount = 4;
+		render_info.pColorAttachments    = &color_attachment_info[0];
+		render_info.renderArea           = {0, 0, width, height};
+
+		render_info.pDepthAttachment = &depth_attachment_info;
+		if (!vkb::is_depth_only_format(depth_format))
+		{
+			render_info.pStencilAttachment = &depth_attachment_info;
+		}
+
+		/*
+		    Dynamic rendering start
+		*/
+		vkCmdBeginRenderingKHR(cmd, &render_info);
+
+		VkViewport viewport = vkb::initializers::viewport(static_cast<float>(width), static_cast<float>(height), 0.0f, 1.0f);
+		vkCmdSetViewport(cmd, 0, 1, &viewport);
+
+		VkRect2D scissor = vkb::initializers::rect2D(width, height, 0, 0);
+		vkCmdSetScissor(cmd, 0, 1, &scissor);
+
+		/*
+		    First draw
+		    Fills the G-Buffer attachments containing image data for the deferred composition (color+depth, normals, albedo)
+		*/
+
+		vkCmdBindPipeline(cmd, VK_PIPELINE_BIND_POINT_GRAPHICS, scene_opaque_pass.pipeline);
+		vkCmdBindDescriptorSets(cmd, VK_PIPELINE_BIND_POINT_GRAPHICS, scene_opaque_pass.pipeline_layout, 0, 1, &scene_opaque_pass.descriptor_set, 0, nullptr);
+		draw_scene(scenes.opaque, cmd, scene_opaque_pass.pipeline_layout);
+
+		// We want to read the input attachments in the next pass, with dynamic rendering local read this requires use of a barrier with the "by region" flag set
+
+		// A new feature of the dynamic rendering local read extension is the ability to use pipeline barriers in the dynamic render pass
+		// to allow framebuffer-local dependencies (i.e. read-after-write) between draw calls using the "by region" flag
+		// So with this barrier we can use the output attachments from the draw call above as input attachments in the next call
+		VkMemoryBarrier2KHR memoryBarrier{};
+		memoryBarrier.sType         = VK_STRUCTURE_TYPE_MEMORY_BARRIER_2_KHR;
+		memoryBarrier.srcStageMask  = VK_PIPELINE_STAGE_2_FRAGMENT_SHADER_BIT;
```

Comment:
Feels like this should be `srcStageMask = VK_PIPELINE_STAGE_2_COLOR_ATTACHMENT_OUTPUT_BIT` and `VK_ACCESS_2_COLOR_ATTACHMENT_WRITE_BIT` in this use-case (unlike the blog, which talks about write/read to storage images -- not write/read of color attachments).

Reply:
This is just a copy of the internal issue for discussion on our next talk. Can you post that comment over there too (Khronos gitlab Vulkan samples repo)?

Reply:
Similarly, is it possible the dstAccessMask here should be `VK_ACCESS_2_COLOR_ATTACHMENT_READ_BIT` instead?

Reply:
Not sure, I think `VK_ACCESS_2_INPUT_ATTACHMENT_READ_BIT` would be correct in here. Will try to have this solved internally by people that know more about this than me ;)

Reply:
Fixed in the gitlab version of the sample.